pkg/controller/operatorsource/factory.go

```diff
@@ -30,24 +27,18 @@ import (
 //
 //  On error, the object is transitioned into "Failed" phase.
 type PhaseReconcilerFactory interface {
-	GetPhaseReconciler(logger *log.Entry, event sdk.Event) (Reconciler, error)
+	GetPhaseReconciler(logger *log.Entry, currentPhase string) (Reconciler, error)
 }
 
 // phaseReconcilerFactory implements PhaseReconcilerFactory interface.
 type phaseReconcilerFactory struct {
 	registryClientFactory appregistry.ClientFactory
 	datastore             datastore.Writer
-	kubeclient            kube.Client
+	client                client.Client
 }
 
-func (s *phaseReconcilerFactory) GetPhaseReconciler(logger *log.Entry, event sdk.Event) (Reconciler, error) {
-	opsrc := event.Object.(*v1alpha1.OperatorSource)
-
-	if event.Deleted {
-		return NewDeletedEventReconciler(logger, s.datastore), nil
-	}
-
```

Comment:
Yeah, controller runtime does not send any `deleted` event - https://github.com/operator-framework/operator-sdk/issues/270.